deepspeed/runtime/pipe/module.py

```diff
@@ -130,6 +130,8 @@ def forward(self, inputs):
         """
 
         super().__init__()
+        if not dist.is_initialized():
```

Comment:
@Quentin-Anthony, please investigate why this is needed? Revert it if its not needed.

Reply:
It was left over from your pipeline-parallel debugging. I removed it.